sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/TinyintTests.java

```diff
@@ -0,0 +1,345 @@
+                            0 |   0
+                           12 |   6
+                          -12 |  -6
+                          127 |  63
+                         -127 | -63
+                        (5 rows)
+
+                        SELECT i.f1, i.f1 / '2'::INT4 AS x FROM INT_TBL i;
+                           f1 |   x
+                        ------+--------
+                            0 |    0
+                           12 |    6
+                          -12 |   -6
+                          127 |   63
+                         -127 |  -63
+                        (5 rows)
+                        
+                        """
+        );
+    }
+
+    // We round down to zero
+    @Test
+    public void testFloatRound() {
+        this.q(
+                """
+                        SELECT x, x::tinyint AS tinyint_value FROM (VALUES (-2.9::float8), (-2.5::float8), (-1.5::float8), (-0.5::float8), (0.0::float8), (0.5::float8), (1.5::float8), (2.5::float8)) t(x);
+                          x   | tinyint_value
+                        ------+------------
+                         -2.9 |         -2
+                         -2.5 |         -2
+                         -1.5 |         -1
+                         -0.5 |          0
+                            0 |          0
+                          0.5 |          0
+                          1.5 |          1
+                          2.5 |          2"""
+        );
+    }
+
+    // We round down to zero
+    @Test
+    public void testNumericRound() {
+        this.q(
+                """
+                        SELECT x, x::tinyint AS tinyint_value FROM (VALUES (-2.9::numeric), (-2.5::numeric), (-1.5::numeric), (-0.5::numeric), (0.0::numeric), (0.5::numeric), (1.5::numeric), (2.5::numeric)) t(x);
+                          x   | tinyint_value
+                        ------+------------
+                         -2.9 |         -2
+                         -2.5 |         -2
+                         -1.5 |         -1
+                         -0.5 |         -0
+                          0.0 |          0
+                          0.5 |          0
+                          1.5 |          1
+                          2.5 |          2"""
+        );
+    }
+
+    // Ignored because this fails in Postgres but here we get:
+    @Test @Ignore("Integer wrapping: https://github.com/feldera/feldera/issues/1186")
+    public void testSelectOverflow() {
+        String error_message = "TINYINT out of range";
+        this.qf("SELECT i.f1, i.f1 * 2::TINYINT AS x FROM INT_TBL i", error_message);
+        this.qf( "SELECT i.f1, i.f1 + '2'::TINYINT AS x FROM INT_TBL i", error_message);
+        this.qf("SELECT i.f1, i.f1 - '2'::TINYINT AS x FROM INT_TBL i", error_message);
+    }
+
+    // This passes for the Calcite version but fails for the run time version
+    @Test @Ignore("Modulo edge case integer overflow: https://github.com/feldera/feldera/issues/1195")
```

Comment:
There is no reason this should fail. There must be a bug in our rust library. The result is 0

Reply:
Yes, this is the same issue as with other integer types. 

https://github.com/feldera/feldera/pull/1196 fixes this.

Reply:
Can we merge 1196?

Reply:
Yes, the merge conflicts have been resolved.